packages/flutter/test/widgets/sensitive_content_test.dart

```diff
@@ -0,0 +1,129 @@
+        );
+
+        expect(find.byWidget(childWidget), findsNothing);
+        final SizedBox shrinkBox = tester.firstWidget(find.byType(SizedBox)) as SizedBox;
+        expect(shrinkBox.width, 0);
+        expect(shrinkBox.height, 0);
+
+        setContentSensitivityCompleter.complete();
+
+        // Delay added to ensure that the SensitiveContent widget registration completes.
+        await Future<void>.delayed(const Duration(milliseconds: 100), () async {
+          await tester.pump();
+        });
+
+        await expectLater(find.byWidget(childWidget), findsOne);
+        expect(find.byType(SizedBox), findsNothing);
+      });
+    },
+  );
+
+  testWidgets(
+    'when SensitiveContent widget changes sensitivity, SizedBox.shrink is built initially, then child widget is built upon completion',
+    (WidgetTester tester) async {
+      int setContentSensitivityCall = 0;
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+        SystemChannels.sensitiveContent,
+        (MethodCall methodCall) {
+          if (methodCall.method == 'SensitiveContent.setContentSensitivity') {
+            setContentSensitivityCall++;
+            // Make second call to update content sensitivity awaits the Future for test.
+            if (setContentSensitivityCall == 2 && methodCall.arguments == 'autoSensitive') {
+              return setContentSensitivityCompleter.future;
+            }
+            return Future<void>.value();
+          } else if (methodCall.method == 'SensitiveContent.getContentSensitivity') {
+            return Future<String>.value(defaultContentSensitivitySetting.name);
+          } else if (methodCall.method == 'SensitiveContent.isSupported') {
+            return Future<bool>.value(true);
+          } else {
+            return null;
+          }
+        },
+      );
+
+      await tester.runAsync(() async {
+        const Key scKey = Key('scKey');
+        final Container childWidget = Container();
+
+        await tester.pumpWidget(
+          ChangeContentSensitivityTester(
+            key: scKey,
+            initialContentSensitivity: ContentSensitivity.sensitive,
+            child: childWidget,
+          ),
+        );
+
+        final ChangeContentSensitivityTesterState scState = tester
+            .firstState<ChangeContentSensitivityTesterState>(find.byKey(scKey));
+        scState.changeContentSensitivityTo(ContentSensitivity.autoSensitive);
+
+        expect(find.byWidget(childWidget), findsNothing);
+        final SizedBox shrinkBox = tester.firstWidget(find.byType(SizedBox)) as SizedBox;
+        expect(shrinkBox.width, 0);
+        expect(shrinkBox.height, 0);
+
+        setContentSensitivityCompleter.complete();
+
+        // Delay added to ensure that the SensitiveContent widget re-registration completes.
+        await Future<void>.delayed(const Duration(milliseconds: 500), () async {
```

Comment:
Do you know where is the async operation this is waiting on? It will be better if we wait for a completer future instead of a delay

Reply:
Ok I thought this was needed to make sure that re-registering the `SensitiveContent` widget happens, but it seems to pass without it. Let's see what happens in CI because I did see a failure before.

We use a completer to complete the fake async method call, but I thought we needed to wait for the `Future` that the `FutureBuilder` relies on to finish to check that the actual `SensitiveContent` widget `child` is built. I thought we couldn't use a `Completer` for this without faking/mocking the `Future` that the `FutureBuilder` relies on.

Reply:
a pump should be enough since unittest uses a fake async. Unless you are introducing scheduleTask or other type a random delay future in the widget, we shouldn't need to wait a real future in unittest. and even then, we should expose a future for test to await in the `await tester.runAsync`